tests/integration-service/integration-with-env.go

```diff
@@ -0,0 +1,170 @@
+	integrationv1beta1 "github.com/konflux-ci/integration-service/api/v1beta1"
+	intgteststat "github.com/konflux-ci/integration-service/pkg/integrationteststatus"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+)
+
+var _ = framework.IntegrationServiceSuiteDescribe("Integration Service E2E tests", Label("integration-service", "integration-env"), func() {
+	defer GinkgoRecover()
+
+	var f *framework.Framework
+	var err error
+
+	var applicationName, componentName, testNamespace string
+	var integrationTestScenario *integrationv1beta1.IntegrationTestScenario
+	var timeout, interval time.Duration
+	var originalComponent *appstudioApi.Component
+	var pipelineRun, integrationPipelineRun *pipeline.PipelineRun
+	var snapshot *appstudioApi.Snapshot
+	var spaceRequest *v1alpha1.SpaceRequest
+
+	AfterEach(framework.ReportFailure(&f))
+
+	Describe("with happy path for general flow of Integration service with ephemeral environment", Ordered, func() {
+		BeforeAll(func() {
+			// Initialize the tests controllers
+			f, err = framework.NewFramework(utils.GetGeneratedNamespace("integration-env"))
+			Expect(err).NotTo(HaveOccurred())
+			testNamespace = f.UserNamespace
+
+			applicationName = createApp(*f, testNamespace)
+			componentName, originalComponent = createComponent(*f, testNamespace, applicationName)
+			integrationTestScenario, err = f.AsKubeAdmin.IntegrationController.CreateIntegrationTestScenario("", applicationName, testNamespace, gitURL, itsRevision, pathIntegrationPipelineWithEnv)
+			Expect(err).ShouldNot(HaveOccurred())
+		})
+
+		AfterAll(func() {
+			if !CurrentSpecReport().Failed() {
+				Expect(f.SandboxController.DeleteUserSignup(f.UserName)).To(BeTrue())
+			}
+		})
+
+		When("a new Component is created", func() {
+			It("triggers a build PipelineRun", Label("integration-service"), func() {
+				pipelineRun, err = f.AsKubeDeveloper.IntegrationController.GetBuildPipelineRun(componentName, applicationName, testNamespace, false, "")
+				Expect(err).ShouldNot(HaveOccurred())
+			})
+
+			It("verifies if the build PipelineRun contains the finalizer", Label("integration-service"), func() {
+				Eventually(func() error {
+					pipelineRun, err = f.AsKubeDeveloper.IntegrationController.GetBuildPipelineRun(componentName, applicationName, testNamespace, false, "")
+					Expect(err).ShouldNot(HaveOccurred())
+					if !controllerutil.ContainsFinalizer(pipelineRun, pipelinerunFinalizerByIntegrationService) {
+						return fmt.Errorf("build pipelineRun %s/%s doesn't contain the finalizer: %s yet", pipelineRun.GetNamespace(), pipelineRun.GetName(), pipelinerunFinalizerByIntegrationService)
+					}
+					return nil
+				}, 1*time.Minute, 1*time.Second).Should(Succeed(), "timeout when waiting for finalizer to be added")
+			})
+
+			It("waits for build PipelineRun to succeed", Label("integration-service"), func() {
+				Expect(pipelineRun.Annotations[snapshotAnnotation]).To(Equal(""))
```

Comment:
I feel if case runs fast, this will fail.

Reply:
you are right @hongweiliu17 ,  but in this test is after that we already had a prior check of the build PLR .